### PR TITLE
[Call Transfer] stabilize transfer

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Stabilizes the transfer feature
# Why

Brings transfer into the stable API

# How Tested

Validated locally. Bug bash to come

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 1]
```diff
@@ -262,8 +262,8 @@ class CallContext {
     const newPage = getCallCompositePage(
       call,
       latestEndedCall,
-      /* @conditional-compile-remove(unsupported-browser) */ environmentInfo,
-      /* @conditional-compile-remove(call-transfer) */ transferCall
+      /* @conditional-compile-remove(call-transfer) */ transferCall,
+      /* @conditional-compile-remove(unsupported-browser) */ environmentInfo
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:266]: Ah ok. Just a re-order will do the trick.

**Review by mgamis-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts (hunk 1 above)
Comment (dmceachernmsft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:266]: yeah this works internal functions only though

**Review by edwardlee-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/pages/CallPage.tsx [hunk 2]
```diff
@@ -105,24 +103,12 @@ export const CallPage = (props: CallPageProps): JSX.Element => {
   /* @conditional-compile-remove(dtmf-dialer) */
   const callees = useSelector(getTargetCallees);
   /* @conditional-compile-remove(dtmf-dialer) */
-  const renderDtmfDialerFromStart = showDtmfDialer(callees);
-
+  const renderDtmfDialerFromStart = showDtmfDialer(callees, remoteParticipantsConnected);
+  console.log('renderDtmfDialerFromStart', renderDtmfDialerFromStart);
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/pages/CallPage.tsx:107]: Should we remove this console log here?

File: packages/react-composites/src/composites/CallComposite/pages/CallPage.tsx [hunk 3]
```diff
@@ -105,24 +103,12 @@ export const CallPage = (props: CallPageProps): JSX.Element => {
...
-   * This useEffect is about clearing the dtmf dialer should there be a new participant that joins the call.
-   * This will only happen the first time should the dialer be present when the call starts.
-   */
-  useEffect(() => {
-    if (remoteParticipantsConnected.length > 1 && dtmfDialerPresent && dialerShouldAutoDismiss.current) {
-      setDtmfDialerPresent(false);
-      dialerShouldAutoDismiss.current = false;
-    }
-  }, [dtmfDialerPresent, remoteParticipantsConnected, setDtmfDialerPresent]);
+  // const dialerShouldAutoDismiss = useRef<boolean>(renderDtmfDialerFromStart);
```
Comment (mgamis-msft) [inline packages/react-composites/src/composites/CallComposite/pages/CallPage.tsx:111]: I think we should remove this commented code

**Pull request merged**